### PR TITLE
feat: added flag for mpv as flatpak

### PR DESCRIPTION
**Pull request by Derisis13**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [x] Feature
- [ ] Documentation update

## Description

Comply with request #1408

## Checklist

- [x] any anime playing
- [ ] bumped version
---
- [x] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--skip` ani-skip works
- [ ] `--skip-title` ani-skip title argument works
- [ ] `--no-detach` no detach works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [x] `-h` help info is up to date
- [ ] Readme is up to date
- [x] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Comment by port19x** [comment]
Didn't we already have that at some point?

**Comment by Derisis13** [comment]
> Didn't we already have that at some point?

We may have. But it seems like there's a need for this. I see no reason why we shouldn't have it.

**Comment by port19x** [comment]
Because we already have a flatpak_mpv player for steamdeck implemented.
I'd rather have [player detection](https://github.com/pystardust/ani-cli/blob/master/ani-cli#L335-L342) fall back to flatpack mpv when a regular mpv can't be detected.
A similar pattern to what I want is already used for iina, where manually installed iina needs different handling than iina installed via homebrew

**Comment by port19x** [comment]
This would have the additional upside of consolidating case branches with steamdeck

**Comment by Derisis13** [comment]
Ok, can do. It makes sense not to not trust it onto our users to know which package format they use

File: ani-cli [hunk 1]
```diff
@@ -337,10 +341,12 @@ quality="${ANI_CLI_QUALITY:-best}"
 case "$(uname -a)" in
     *Darwin*) player_function="$(where_iina)" ;;                      # mac OS
     *ndroid*) player_function="${ANI_CLI_PLAYER:-android_mpv}" ;;     # Android OS (termux)
-    *neptune*) player_function="${ANI_CLI_PLAYER:-flatpak_mpv}" ;;    # steamdeck OS
     *MINGW* | *WSL2*) player_function="${ANI_CLI_PLAYER:-mpv.exe}" ;; # Windows OS
     *ish*) player_function="${ANI_CLI_PLAYER:-iSH}" ;;                # iOS (iSH)
-    *) player_function="${ANI_CLI_PLAYER:-mpv}" ;;                    # Linux OS
+    *) 
+      player_function="${ANI_CLI_PLAYER:-$(where_mpv)}"               # Linux OS
+      [ $? -eq 0 ] || exit $?
```
Comment (port19x) [inline ani-cli:348]: This part is actually not necessary, since `dep_ch` will already exit

**Review by port19x — CHANGES_REQUESTED** [review]

**Comment by Derisis13** [comment]
I've tested what I could on linux, but @ykhan21 I'd like you to test it on windows if it correctly detects installed and uninstalled mpv, and @Nannk to check it on the steamdeck if it still finds the flatpak version.
Also @port19x I think there's a little issue with `where_iina()` which I couldn't test: when the subshell exits, the main program continues uninterrupted. I suggest to use a similar structure like I did with `where_mpv()`

File: ani-cli (hunk 1 above)
Comment (Derisis13) [reply ani-cli:348]: The whole reason I did that so we can detect if either packaging format has mpv. I'm open to suggestions for better options.

**Comment by port19x** [comment]
iirc it worked fine in my testing. `return` exits subshells, `exit` always aborts the shell script.

[SO Post](https://stackoverflow.com/questions/4419952/difference-between-return-and-exit-in-bash-functions)

Where is this not the case?

**Comment by port19x** [comment]
Could it be because you use a default assignment? `${ANI_CLI_PLAYER:-$(where_mpv)}`

In where iina, that is checked within the function, but I'd be surprised if that changes things

**Comment by Derisis13** [comment]
> iirc it worked fine in my testing on mac os. `return` exits subshells, `exit` always aborts the shell script.
> 
> [SO Post](https://stackoverflow.com/questions/4419952/difference-between-return-and-exit-in-bash-functions)
> 
> Where is this not the case?

Linux and dash. Same if the default assignment is not present, like with iina.

**Comment by Derisis13** [comment]
> iirc it worked fine in my testing on mac os. `return` exits subshells, `exit` always aborts the shell script.
> 
> [SO Post](https://stackoverflow.com/questions/4419952/difference-between-return-and-exit-in-bash-functions)
> 
> Where is this not the case?

Linux and dash. Same if the default assignment is not present, like with iina.

And checking the [shell specification's related section](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#exit) reveals that `exit` only returns from the current subshell and continues from the environment it was called from.

**Comment by port19x** [comment]
Alright, I'm taking the L

**Comment by Derisis13** [comment]
Now then why haven't the two remaining checks ran?

**Comment by port19x** [comment]
> Now then why haven't the two remaining checks ran?

I have no idea. I added the completion files to path ignore, but you haven't changed those in this PR, so I'm surprised they aren't running. Github Actions is a bit fragile at times.

Wanna patch my where_iina to be more in line with your where_mpv?

Let's see if github wakes from its slumber after another commit

**Comment by Derisis13** [comment]
> Wanna patch my where_iina to be more in line with your where_mpv?

Sure, let's see

**Comment by port19x** [comment]
Now that's a surprise. Perhaps a brown out on githubs part to bully people into migrating away from a deprecated nodejs version (used by checkout v3)

**Review by port19x — APPROVED** [review]

**Pull request merged**